pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -388,41 +387,35 @@ func (e *aksOperatorController) deployAKSOperator() error {
 }
 
 func (e *aksOperatorController) generateSATokenWithPublicAPI(cluster *mgmtv3.Cluster) (string, *bool, error) {
```

Comment:
Could all KEv2 clusters use the same `generateSATokenWIthPublicAPI` if the restConfig were passed as a parameter?